src/stronghold/secret.rs

```diff
@@ -36,35 +38,29 @@ impl SecretManage for StrongholdAdapter {
         _metadata: GenerateAddressMetadata,
     ) -> Result<Vec<Address>> {
         // Stronghold arguments.
-        let seed_location = SLIP10DeriveInput::Seed(Location::Generic {
-            vault_path: SECRET_VAULT_PATH.to_vec(),
-            record_path: SEED_RECORD_PATH.to_vec(),
-        });
-        let derive_location = Location::Generic {
-            vault_path: SECRET_VAULT_PATH.to_vec(),
-            record_path: DERIVE_OUTPUT_RECORD_PATH.to_vec(),
-        };
-        let hint = RecordHint::new(RECORD_HINT).unwrap();
+        let seed_location = Slip10DeriveInput::Seed(Location::generic(SECRET_VAULT_PATH, SEED_RECORD_PATH));
+        let derive_location = Location::generic(SECRET_VAULT_PATH, DERIVE_OUTPUT_RECORD_PATH);
 
         // Addresses to return.
         let mut addresses = Vec::new();
 
         for address_index in address_indexes {
-            // Stronghold 0.4.1 is still using an older version of iota-crypto, so we construct a different one here.
-            let chain = crypto05::keys::slip10::Chain::from_u32_hardened(vec![
-                44u32,
-                coin_type,
-                account_index,
-                internal as u32,
-                address_index,
-            ]);
+            let chain = Chain::from_u32_hardened(vec![44u32, coin_type, account_index, internal as u32, address_index]);
+
+            let stronghold_client = self.stronghold.lock().await.get_client(PRIVATE_DATA_CLIENT_PATH)?;
 
             // Derive a SLIP-10 private key in the vault.
-            self.slip10_derive(chain, seed_location.clone(), derive_location.clone(), hint)
-                .await?;
+            stronghold_client.execute_procedure(procedures::Slip10Derive {
```

Comment:
If we call the procedure like this now, we can remove `slip10_derive()` in secret.rs? Same for `ed25519_public_key()`

Reply:
I think I actually prefer the methods so maybe I'll replace the procedures with them